pkg/analysis_server/test/services/correction/fix_test.dart

```diff
@@ -6139,6 +6139,134 @@ class A {
 ''');
   }
 
+  test_removeTypeName_avoidAnnotatingWithDynamic_NamedParameter() async {
+    String src = '''
+bad({/*LINT*/dynamic defaultValue}) {
+  return null;
+}
+''';
+    await findLint(src, LintNames.avoid_annotating_with_dynamic);
+
+    await applyFix(DartFixKind.REMOVE_TYPE);
+
+    verifyResult('''
+bad({defaultValue}) {
+  return null;
+}
+''');
+  }
+
+  test_removeTypeName_avoidAnnotatingWithDynamic_NormalParameter() async {
+    String src = '''
+bad(/*LINT*/dynamic defaultValue) {
+  return null;
+}
+''';
+    await findLint(src, LintNames.avoid_annotating_with_dynamic);
+
+    await applyFix(DartFixKind.REMOVE_TYPE);
+
+    verifyResult('''
+bad(defaultValue) {
+  return null;
+}
+''');
+  }
+
+  test_removeTypeName_avoidAnnotatingWithDynamic_OptionalParameter() async {
+    String src = '''
+bad([/*LINT*/dynamic defaultValue]) {
+  return null;
+}
+''';
+    await findLint(src, LintNames.avoid_annotating_with_dynamic);
+
+    await applyFix(DartFixKind.REMOVE_TYPE);
+
+    verifyResult('''
+bad([defaultValue]) {
+  return null;
+}
+''');
+  }
```

Comment:
Nope, we decided that for now it was only with parameters, we would like to extend (the lint) to return type also but we decided that not yet because omit them could not maintain the same behavior than typing dynamic